### ISSUE TITLE
error in error message

### ISSUE THREAD

**Issue by tsgouros** [ivlab/MinVR2#179]
This code isn't right, (VRSharedLibrary.cpp).  Needs to have the error variable set if the LoadLibraryA fails. I'm guessing this is a DanO thing?

void VRSharedLibrary::load() {
	if (!_isLoaded) {

		const char* error;

#if defined(WIN32)
		_lib = LoadLibraryA(_filePath.c_str());
#else
		dlerror();
		_lib = dlopen(_filePath.c_str(), RTLD_NOW);//RTLD_LAZY);
		error = dlerror();
#endif

		if (!_lib) {

#ifdef MinVR_DEBUG
			VRWARNING("Could not load library: " + _filePath + " - " + error,
                "This is a harmless warning, unless the library can't be found anywhere else.");
#endif
			return;

		}

    _isLoaded = true;

    DEBUGMSG("Loaded library: " + _filePath)
	}
}

**Comment by dtorban** [comment]
Hey Tom, can you explain more what the problem is?  I don't completely understand.

**Comment by tsgouros** [comment]
Hi Dan:  It's just a coding error. If there is an error when MinVR_DEBUG is enabled, the `error` variable will not have been initialized when the error is issued. I don't know enough about LoadLibraryA() to know how to fill it in, thought you would. I see I messed up the formatting above. How's this?

```
void VRSharedLibrary::load() {
  if (!_isLoaded) {
    const char* error;
#if defined(WIN32)
    _lib = LoadLibraryA(_filePath.c_str());
#else
    dlerror();
    _lib = dlopen(_filePath.c_str(), RTLD_NOW);//RTLD_LAZY);
    error = dlerror();
#endif

    if (!_lib) {
#ifdef MinVR_DEBUG
    VRWARNING("Could not load library: " + _filePath + " - " + error,
            "This is a harmless warning, unless the library can't be found anywhere else.");
#endif
    return;
}
```

**Comment by tsgouros** [comment]
This is still a problem. Someone else want to take a crack at it? Does LoadLibraryA provide an error message? Google isn't helping me figure that out.

**Comment by dtorban** [comment]
I will take a look.

Thanks,
Dan

**Comment by tsgouros** [comment]
Can someone take this on? All it needs is an "error = ???" statement after loadLibraryA().

Thanks,

 -Tom

**Comment by tsgouros** [comment]
Looks like this is addressed with #231
